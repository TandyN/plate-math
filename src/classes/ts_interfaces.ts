enum WeightType {
  kgs = 'kilograms',
  lbs = 'pounds',
}

export { WeightType }
