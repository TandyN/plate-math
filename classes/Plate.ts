import { WeightType } from './ts_interfaces'

const plate_images: { [key in WeightType]: { [key: number]: string } } = {
  kilograms: {
    0.25: 'img',
    0.5: 'img',
    1.25: 'img',
    2.5: 'img',
    5: 'img',
    10: 'img',
    15: 'img',
    20: 'img',
    25: 'img',
  },
  pounds: {
    1.25: 'img',
    2.5: 'img',
    5: 'img',
    10: 'img',
    25: 'img',
    35: 'img',
    45: 'img',
  },
}

class Plate {
  weight: number
  weight_type: WeightType
  image: string

  constructor(weight: number, weight_type: WeightType) {
    if (weight < 0) {
      throw new Error('plate weight can not be negative')
    }

    if (!Object.values(WeightType).includes(weight_type)) {
      console.warn(
        `weight_type '${weight_type}' does not exist for plate. assuming kilograms`,
      )
      weight_type = WeightType.kgs
    }

    this.weight = weight
    this.weight_type = weight_type

    if (!plate_images[weight_type][weight]) {
      throw new Error(
        `plate number '${weight}' does not exist for '${weight_type}'`,
      )
    }

    this.image = plate_images[weight_type][weight]
  }
}

export { Plate }
