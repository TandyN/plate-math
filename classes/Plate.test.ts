import { Plate } from './Plate'
import { WeightType } from './ts_interfaces'

describe('Plate Class', () => {
  it('should throw an error if plate number does not exist', () => {
    expect(() => {
      new Plate(45, WeightType.kgs)
    }).toThrowErrorMatchingSnapshot()
  })

  it('should throw an error if plate type does not exist', () => {
    expect(() => {
      new Plate(45, 'pound' as never)
    }).toThrowErrorMatchingSnapshot()
  })

  it('should insert the correct weight & weight type', () => {
    const weight = new Plate(45, WeightType.lbs)

    expect(weight.weight).toBe(45)
    expect(weight.weight_type).toBe('pounds')
    expect(typeof weight.plate_image).toBe('string')
  })
})
