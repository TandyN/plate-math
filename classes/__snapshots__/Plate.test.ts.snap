// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plate Class should throw an error if initializing plate with negative number 1`] = `"plate weight can not be negative"`;

exports[`Plate Class should throw an error if initializing plate with negative number 2`] = `"plate weight can not be negative"`;

exports[`Plate Class should throw an error if plate number does not exist for kilograms 1`] = `"plate number '46' does not exist for 'kilograms'"`;

exports[`Plate Class should throw an error if plate number does not exist for pounds 1`] = `"plate number '46' does not exist for 'pounds'"`;

exports[`Plate Class should throw error if weight type does not exist when intializing plate 1`] = `"weight type 'fake weight type' does not exist for plate"`;
